src/CartesianProduct.jl

```diff
@@ -727,3 +765,42 @@ function substitute_blocks(low_dim_cpa::CartesianProductArray{N},
     end
     return CartesianProductArray(array)
 end
+
+"""
+    linear_map(M::AbstractMatrix{N}, cpa::CartesianProductArray{N}
+              ) where {N<:Real}
+
+Concrete linear map of a Cartesian product of a finite number of convex sets.
+
+### Input
+
+- `M`   -- matrix
+- `cpa` -- Cartesian product of a finite number of convex sets
+
+### Output
+
+A polytope.
+
+### Algorithm
+
+We check if the matrix is invertible.
+If so, we convert the Cartesian product to constraint representation.
+Otherwise, we convert the Cartesian product to vertex representation.
+In both cases, we then call `linear_map` on the resulting polytope.
+"""
+function linear_map(M::AbstractMatrix{N}, cpa::CartesianProductArray{N}
```

Comment:
This function looks similar to the one for `CartesianProduct`. Using a `Union` type in the previous function may avoid this code dup.

Reply:
True. I was thinking about it, but we usually created two versions here. The main reason why I created two methods even if the code is identical was that we would have to decide where to put the docs.
But I could use meta programming to create the two functions in a loop. How does that sound?

Reply:
I tried but could not figure out how to document *methods* programmatically. [The Julia docs](https://docs.julialang.org/en/v1/manual/documentation/#Advanced-Usage-1) only explain how to document *functions*. [The text here](https://github.com/JuliaLang/julia/blob/b5c4e63db08a56f442b35ffd85a419c54c9c0cc1/base/docs/Docs.jl#L55-L59) seems to indicate that this is not actually possible. So I ended up outsourcing the code to a helper function instead.